### ISSUE TITLE
Multiple classes (wrapped objects)

### ISSUE THREAD

**Issue by johnsauer99** [nodejs/node-addon-examples#180]
My problem is multiple classes (objects) exported via my Init() functions.
I have two classes that I want to use in my javascript but the 'constructor' seems to be overwritten with whoever gets Init()'d i.e.:

// This code is present in both classes Init() method:
    Napi::FunctionReference* constructor = new Napi::FunctionReference();
    *constructor = Napi::Persistent(func);
    env.SetInstanceData(constructor);
    exports.Set("Dataset", func);   

Question: Is it possible to export have more than one class object??

**Comment by mhdawson** [comment]
I'm not quite sure what you mean by "class" in this context.  There should only be one Napi::Addon instance for a given addon. 

You can create multiple Napi::Object wrap instances.

**Comment by ikokostya** [comment]
Related issue https://github.com/nodejs/node-addon-examples/issues/171

**Comment by ikokostya** [comment]
@mhdawson All examples repeat the following code snippet:

```cpp
Napi::FunctionReference* constructor = new Napi::FunctionReference();
*constructor = Napi::Persistent(func);
env.SetInstanceData(constructor);
```

So there is only one data item can be associated with the current instance of the addon. In our case only one constructor. This works only for examples which contain single object wrap. Please see #171 for more details.

**Comment by mhdawson** [comment]
You are correct that only one data item can be associated with the current instance. Y

You can make what you set through env.SetIntsanceData a struct so that it can hold multiple C level pointers. Or possibly a linked list. If you make it a linked list then you could use GetInstanceData to get the current value set. If it is Null create the initial entry and set it, if it is not null then add to the linked list.  At shutdown you'd then need to walk the list and do the right thing for each entry.
